app/models/account_suggestions.rb

```diff
@@ -1,28 +1,49 @@
 # frozen_string_literal: true
 
 class AccountSuggestions
+  include DatabaseHelper
+
   SOURCES = [
     AccountSuggestions::SettingSource,
-    AccountSuggestions::PastInteractionsSource,
+    AccountSuggestions::FriendsOfFriendsSource,
+    AccountSuggestions::SimilarProfilesSource,
     AccountSuggestions::GlobalSource,
   ].freeze
 
-  def self.get(account, limit)
-    SOURCES.each_with_object([]) do |source_class, suggestions|
-      source_suggestions = source_class.new.get(
-        account,
-        skip_account_ids: suggestions.map(&:account_id),
-        limit: limit - suggestions.size
-      )
+  BATCH_SIZE = 40
 
-      suggestions.concat(source_suggestions)
-    end
+  def initialize(account)
+    @account = account
   end
 
-  def self.remove(account, target_account_id)
-    SOURCES.each do |source_class|
-      source = source_class.new
-      source.remove(account, target_account_id)
+  def get(limit, offset = 0)
+    with_read_replica do
+      account_ids_with_sources = Rails.cache.fetch("follow_recommendations/#{@account.id}", expires_in: 15.minutes) do
+        SOURCES.each_with_object([]) do |klass, arr|
+          source = klass.new
+          arr.concat(source.get(@account, limit: BATCH_SIZE).zip([source.key].cycle))
+        end.shuffle
```

Comment:
Something like this sounds cleaner:
```suggestion
        SOURCES.flat_map do |klass|
          klass.new.get(@account, limit: BATCH_SIZE).zip([source.key].cycle)
        end.shuffle
```